src/Error/Error.php

```diff
@@ -47,28 +47,43 @@
     private $filePath;
 
     /**
-     * @param int    $type
-     * @param string $filePath
+     * @var \Exception|\Error|\Throwable|null
      */
-    public function __construct($type, $filePath)
+    private $source;
+
+    /**
+     * @param int                               $type
+     * @param string                            $filePath
+     * @param \Exception|\Error|\Throwable|null $source
```

Comment:
As far as I understand, [ABNF definition in proposed PSR-5](https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md#abnf) does not seem to allow this syntax. Is it supported by common IDEs like PhpStorm?